### ISSUE TITLE
Different order on window and macOS on adoc Generation 

### ISSUE THREAD

**Issue by wetzesa** [asciidoctor/asciidoctor-maven-plugin#552]
I had today a strange behavior. The PDF generation on macOS works but during the generation on Windows 10, a included fragment, overrides all content from the index.adoc and at the end the PDF is empty.

Now the details:
The small document is splitted in 2 adoc files and generated with Maven 3.8.2 and AdoptOpenJDK 11.0.12 on both platforms.
The index.adoc is in the root folder next to the pom.xml and contain the content.
The metadata.adoc, as shared fragment, in the folder "shared/fragments/metadata.adoc" contains stuff like global variables or default settings for the same style, etc.

On macOS I rececive a normal PDF multipage document with content after the Maven run. It is what i want ;-)
But on Windows the document is only a one pager and empty. Only few static thinks in the footer are there.

I checked the Commandline output and there is the different, that on macOS first the metadata.adoc and than the index.adoc will be generated.
On Windows it is reverse. It's generate first the index.adoc and after that the metadata.adoc which overrides all.

Anyone an idea what is wrong there or is there any config where I can set the order of generation?

More details about the setup / version infos:
<asciidoctor.maven.plugin.version>2.2.1</asciidoctor.maven.plugin.version>
<asciidoctorj.pdf.version>1.6.0</asciidoctorj.pdf.version>
<asciidoctorj.version>2.5.2</asciidoctorj.version>
<jruby.version>9.2.19.0</jruby.version>

**Comment by wetzesa** [comment]
I would, but my windows VM is not fully setuped because is it only a test env for this issue. Or is ist a single command without any other dependencies?
In the past I generated only with the IDE or with Maven.

**Comment by abelsromero** [comment]
I could reproduce the order and that makes me think the problem is in the configuration.
@wetzesa I assume you are overriding source's path and more importantly the `outputFile`.

```xml
    <sourceDirectory>${project.basedir}</sourceDirectory>
    <outputFile>output.pdf</outputFile>
```

If that's the case, each AsciiDoc source found is converted and written to the same file, so in MacOs it works because index is the last one being converted (fun fact, Windows and Linux share same order). A warning like this should appear in the output too:

```
[WARNING] Duplicated destination found: overwriting file: ____/asciidoctor-pdf-example/target/generated-docs/output.pdf
```

As as a suggestion, is good to narrow the plugin to what you really need to convert. This helps performance avoiding unnecessary file traversing and conversions.
* If you are converting a single file, use `sourceDocumentName` to point to your "index" document. That way nothing else is converted.
* If you have multiple files, "hide" included sources with "_" prefix: included files with "_" (eg. `_metadata.adoc`), or  folders (e.g. `_shared`) can be included from other sources but are ignored when looking for sources.

Also, I'd highly recommend not placing docs in the root if possible.

**Comment by wetzesa** [comment]
Oh man...you are my salvation!
It is exact the case you are writing about. I added now `sourceDocumentName` and now it works. The duplication warning is also gone. Really nice ;-)

About your suggestion to narrow to what is necessary...
Yes it is correct. I'm not complete familiar with asciidoc and it sounds right.
I have changed both. `sourceDocumentName` and also the includes.
On the first few generations after the changes on my small project the improvment is not noticeable.
But I will do it correct and follow your suggestion!

Thx much!
